src/Razor/test/RazorLanguage.Test/IntegrationTests/ComponentCodeGenerationTestBase.cs

```diff
@@ -998,6 +998,24 @@ public void BuiltIn_BindToInputWithoutType_WritesAttributes()
             CompileToAssembly(generated);
         }
 
+        [Fact]
+        public void BuiltIn_BindToInputWithoutType_IsCaseSensitive()
+        {
+            // Arrange
+
+            // Act
+            var generated = CompileToCSharp(@"
+<input @BIND=""@ParentValue"" />
+@code {
+    public int ParentValue { get; set; } = 42;
+}");
+
+            // Assert
+            AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
+            AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
+            CompileToAssembly(generated);
```

Comment:
Should this have a diagnostic? DO we end up treating it as an expression in this case?

Reply:
Yeah we just treat it like a regular attributes with an expression value. I'm not sure if we want to have a diagnostic every time a tag helper descriptor could match but can't because of casing.